### DISCUSSION TITLE
on_ready inside extension

### DISCUSSION THREAD

**Discussion by ahmedbkd** [Rapptz/discord.py#6551] [Q&A] [answered]
So I want to run a async function inside extension which is something equivalent to _`on_ready()`_ but inside the extension, like when the extension is completely loaded. So I can run the function using await keyword because I can't use await inside the constructor, because you can't use await outside async function

Sorry for my bad explanation.. any ideas?

**Comment by Kellenn** [comment, marked as answer]
You can run an asynchronous method from ``__init__`` by using [loop.create_task](https://docs.python.org/3/library/asyncio-eventloop.html?highlight=create_task#asyncio.loop.create_task).
``create_task`` Schedules a task on your bot's event loop.
```py
from discord.ext import commands


class MyCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        bot.loop.create_task(self.on_load())

    async def on_load(self):
        print(f'{self.qualified_name} loaded')

def setup(bot):
    bot.add_cog(MyCog(bot))
```
Additionally, you can always use ``on_ready``; Keep in mind this event is not guaranteed to be called only once.

```py
from discord.ext import commands


class MyCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.listener()
    async def on_ready(self):
        print(f'{self.qualified_name} ready')

def setup(bot):
    bot.add_cog(MyCog(bot))
```
